### PR TITLE
fix: violations for React Rules Of Hooks (essentially: non-conditional, exhaustive dependency arrays)

### PR DESCRIPTION
**Pull request by danielweck**
Current lint status (`develop` branch, all fixed by this PR but needs reviewing and breaking down into smaller commits, for easier rollback if necessary):

`src/renderer/common/hooks/useApi.ts`
  30:8  error  React Hook React.useEffect has missing dependencies: 'requestId' and 'store'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

`src/renderer/common/hooks/useTranslator.ts`
  23:8  error  React Hook React.useEffect has a missing dependency: 'translator'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

`src/renderer/library/components/dialog/DeleteOpdsFeedConfirm.tsx`
  27:8  error  React Hook React.useCallback has a missing dependency: 'dispatch'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

`src/renderer/library/components/dialog/DeletePublicationConfirm.tsx`
  28:8  error  React Hook React.useCallback has a missing dependency: 'dispatch'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

`src/renderer/library/components/searchResult/AllPublicationPage.tsx`
   526:8  error  React Hook React.useEffect has missing dependencies: 'onInputChange', 'props.accessibilitySupportEnabled', 'props.column', and 'searchParams'. Either include them or remove the dependency array                                                                                                                            react-hooks/exhaustive-deps
  1617:8  error  React Hook React.useMemo has missing dependencies: '__' and 'props'. Either include them or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useMemo call and refer to those specific props inside React.useMemo  react-hooks/exhaustive-deps
  1864:8  error  React Hook React.useMemo has a missing dependency: 'props'. Either include it or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useMemo call and refer to those specific props inside React.useMemo             react-hooks/exhaustive-deps

`src/renderer/reader/components/ReaderMenu.tsx`
   604:8  error  React Hook React.useEffect has a missing dependency: 'pageNumber'. Either include it or remove the dependency array                                                                                                                           react-hooks/exhaustive-deps
   608:8  error  React Hook React.useEffect has a missing dependency: 'getStartPage'. Either include it or remove the dependency array                                                                                                                         react-hooks/exhaustive-deps
   943:9  error  React Hook React.useMemo has a missing dependency: 'r2Publication.Spine'. Either include it or remove the dependency array                                                                                                                    react-hooks/exhaustive-deps
   953:8  error  React Hook React.useEffect has a missing dependency: 'pageNumber'. Either include it or remove the dependency array. You can also do a functional update 'setPageNumber(p => ...)' if you only need 'pageNumber' in the 'setPageNumber' call  react-hooks/exhaustive-deps
  1519:8  error  React Hook React.useEffect has a missing dependency: 'dockingMode'. Either include it or remove the dependency array                                                                                                                          react-hooks/exhaustive-deps

`src/renderer/reader/components/ReaderSettings.tsx`
   303:8  error  React Hook React.useEffect has a missing dependency: 'selectFont'. Either include it or remove the dependency array                                                                     react-hooks/exhaustive-deps
  1085:8  error  React Hook React.useMemo has missing dependencies: 'setSettings' and 'setTranscientStateOverridePublisherDefault'. Either include them or remove the dependency array                   react-hooks/exhaustive-deps
  1117:8  error  React Hook React.useMemo has missing dependencies: 'setOverridePublisherDefault', 'setPartialSettingsDebounced', and 'setTabValue'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

### CODE REVIEW

**Comment by danielweck** [comment]
Hi @panaC I merged to `develop` the non-controversial (obvious) fixes, such as eliminating conditional hooks.
This PR now contains the fixes for all the specific violations of React Rules of Hooks / exhaustive dependencies.
However this PR is not ready to merge yet, because I want to thoroughly check potential excessive re-renders and incorrect memoizations / cache invalidations (particularly with function dependencies / useCallback).
In fact, when reviewing the codebase to create this PR (with the help of the official React linter, of course) I discovered several code paths that clearly posed risks of unnecessary re-renders, so the linter is helping us not only write "correct" React code but also make sure that our render optimizations are effective.
Let's talk when you have time :)

File: src/renderer/common/hooks/useApi.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export function useApi<T extends TApiMethodName>(_requestId: string | undefined,
         return () => {
             store.dispatch(apiActions.clean.build(requestId));
         };
-    }, []);
+    }, [requestId, store]);
```
Comment (panaC) [inline src/renderer/common/hooks/useApi.ts:30]: seems good

Comment (panaC) [reply src/renderer/common/hooks/useApi.ts:30]: the clean dispatch occurs at the unmount of the component with the last requestId

File: src/renderer/common/hooks/useTranslator.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function useTranslator(): [typeof Translator.prototype.translate, Transla
             forceUpdate();
         };
         return translator.subscribe(handleLocaleChange);
-    }, [translator.subscribe]);
```
Comment (panaC) [inline src/renderer/common/hooks/useTranslator.ts:23]: seems logic

File: src/renderer/library/components/dialog/DeleteOpdsFeedConfirm.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ const DeleteOpdsFeedConfirm = (props: { feed: IOpdsFeedView, trigger: React.Reac
     const removeAction = React.useCallback(() => {
         dispatch(dialogActions.closeRequest.build());
         remove(props.feed.identifier);
-    }, [remove, props.feed.identifier]);
+    }, [dispatch, remove, props.feed.identifier]);
```
[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/library/components/dialog/DeleteOpdsFeedConfirm.tsx:27]: In reality this is not the case

File: src/renderer/library/components/dialog/DeletePublicationConfirm.tsx [hunk 4]
```diff
@@ -25,7 +25,7 @@ const DeletePublicationConfirm = (props: { publicationView: PublicationView, tri
     const removeAction = React.useCallback(() => {
         dispatch(dialogActions.closeRequest.build());
         remove(props.publicationView.identifier);
-    }, [remove, props.publicationView.identifier]);
+    }, [remove, props.publicationView.identifier, dispatch]);
```
Comment (panaC) [inline src/renderer/library/components/dialog/DeletePublicationConfirm.tsx:28]: ok

File: src/renderer/common/hooks/useApi.ts (hunk 1 above)
Comment (danielweck) [reply src/renderer/common/hooks/useApi.ts:30]: yes, `requestId` is already a "first mount" useMemo, and `store` is a useContext "constant" (so to speak) ... so the behaviour with `[]` empty deps array is identical ... looks like a safe refactor that satisfies the compiler / linter

File: src/renderer/library/components/dialog/DeleteOpdsFeedConfirm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielweck) [reply src/renderer/library/components/dialog/DeleteOpdsFeedConfirm.tsx:27]: in practice `dispatch` is a "constant" but from the compiler / linter's perspective, this is a reactive value that can mutate during the component lifecycle, thereby triggering a hook re-run

File: src/renderer/reader/components/ReaderMenu.tsx [hunk 5]
```diff
@@ -578,7 +578,7 @@ const annotationCardContext = React.createContext<{
 
 const AnnotationList: React.FC<{ r2Publication: R2Publication, dockedMode: boolean, annotationUUIDFocused: string, doFocus: number} & Pick<IReaderMenuProps, "goToLocator">> = (props) => {
```
Comment (panaC) [inline src/renderer/reader/components/ReaderMenu.tsx:579]: `doFocus` is not used anymore

Comment (panaC) [reply src/renderer/reader/components/ReaderMenu.tsx:579]: just to `log` it at re-render

File: src/renderer/common/hooks/useApi.ts (hunk 1 above)
Comment (danielweck) [reply src/renderer/common/hooks/useApi.ts:30]: note: React strict mode, effects run twice in dev mode

File: src/renderer/reader/components/ReaderMenu.tsx (hunk 5 above)
Comment (panaC) [reply src/renderer/reader/components/ReaderMenu.tsx:579]: it's a tricks to re-render the component when Reader.tsx request it

**Comment by panaC** [comment]
seems good for me, need to check with the #2438

**Pull request merged**